### PR TITLE
Refactor to use key-master/gate-keeper

### LINKED ISSUE #6: Break `currentlyRefreshing` logic out to its own module
https://github.com/TehShrike/levelup-cache/blob/53f517e6c81dcad0d13bd49e67ace4d7d925308e/index.js#L33
- New module should be initialized with an async getter function
- Should expose an `isCurrentlyGetting` method
- Should expose a `cancelGet` method
- Should only allow one async `get` call at a time, and pass any arguments given to the callback along to all previous requestors when it returns

When used with the key-master this should remove a good bit of code, and more importantly, the difficult async code that always makes me think too hard when I look at this module's code.

Depends on https://github.com/TehShrike/key-master/issues/4

### PR DESCRIPTION
**Pull request by TehShrike**
Now I'm wondering if I should just use the es6 browserify transform too and clean it up even more with arrow functions

Fixes #6

### CODE REVIEW

**Comment by TehShrike** [comment]
I should specify [engine](https://docs.npmjs.com/files/package.json#engines) in this module and also gate-keeper and key-master and every other module of mine

**Comment by TehShrike** [comment]
Any thoughts @ArtskydJ or @saibotsivad?

**Comment by TehShrike** [comment]
I took the plunge and dropped support for anything older than node 6 from this module, and from gate-keeper.

File: package.json [hunk 1]
```diff
@@ -3,27 +3,48 @@
...
   "main": "index.js",
+  "engines": {
+    "node": ">=6.0.0"
+  },
   "directories": {
     "test": "test"
   },
   "devDependencies": {
     "asynquence": "0.8.2",
+    "babel-preset-es2015": "6.14.0",
```
Comment (ArtskydJ) [inline package.json:14]: Do you need this if you require node@6?

File: .travis.yml [hunk 2]
```diff
@@ -1,4 +1,11 @@
 language: node_js
 node_js:
-  - "0.11"
-  - "0.10"
+  - "6"
+addons:
+  apt:
+    packages:
+      - xvfb
+install:
```
Comment (ArtskydJ) [inline .travis.yml:8]: Why do you need xvfb?

**Comment by TehShrike** [comment]
The main benefit here is `index.js` losing 14 lines, specifically the lines having to do with checking/changing `currentlyRefreshing`.  That logic is all inside [gate-keeper](https://github.com/TehShrike/gate-keeper) now, with its own tests.

File: .travis.yml (hunk 2 above)
Comment (TehShrike) [reply .travis.yml:8]: So [tape-run](https://github.com/juliangruber/tape-run) can run Electron on travis.

File: package.json (hunk 1 above)
Comment (TehShrike) [reply package.json:14]: Yeah - my goal is to support two use cases: node via node 6, and ES5-compatible browsers via browserify.  Babel is needed to support the latter.

**Pull request merged**